tuf/data/snapshot.go

```diff
@@ -23,6 +24,27 @@ type Snapshot struct {
 	Meta    Files     `json:"meta"`
 }
 
+// isValidSnapshot returns an error, or nil, depending on whether the content of the struct
+// is valid for snapshot metadata.  This does not check signatures or expiry, just that
+// the metadata content is valid.
+func isValidSnapshot(s Snapshot) error {
+	expectedType := TUFTypes[CanonicalSnapshotRole]
+	if s.Type != expectedType {
+		return ErrInvalidMeta{
+			Role: CanonicalSnapshotRole, Msg: fmt.Sprintf("expected type %s, not %s", expectedType, s.Type)}
+	}
+
+	for _, role := range []string{CanonicalRootRole, CanonicalTargetsRole} {
```

Comment:
Actually after discussion with @cyli: we may not want to validate names of roles other than root and targets, in case we want to chain previous snapshots.